### ISSUE TITLE
Ensure that we can run tests via common.tests.zip even without a `test_packages_url` available

### ISSUE THREAD

**Issue by whimboo** [mozilla/mozmill-ci#738]
Right now we run our fx-ui-tests on staging via the common.tests.zip package. There is no need for the github repository anymore. The problem is that this works for functional tests on en-US builds only. But not for other locales or even update tests. The main reason is that the pulse messages we receive do not contain this URL.

Here my idea:
1. For other locales we can try to use mozdownload to fetch the appropriate package url from the en-US build with the same buildid. This is hopefully trivial.

2. For update tests we have to account for that at least two different builds will be used. Given that in newer builds we can have fallback code, we should try to get the URL from the target build. This can be done the same way as 1)

**Comment by whimboo** [comment]
Before we can have this feature for release builds I have to get https://github.com/mozilla/mozdownload/issues/354 implemented.

**Comment by whimboo** [comment]
mozdownload 1.20.1 has been released with the necessary new feature. So I can get started on this issue.

**Comment by whimboo** [comment]
Got my first job running for a beta ar locale build via test packages as picked by a parent tinderbox build:

https://treeherder.allizom.org/#/jobs?repo=mozilla-beta&revision=66afdb3943e9&filter-tier=1&filter-tier=2&filter-tier=3&selectedJob=608757

I will finalize the PR when I'm back from PTO.

**Comment by whimboo** [comment]
PR #748 got merged. I will push it to staging and check the next days how it works.

**Comment by whimboo** [comment]
So this doesn't work as I just noticed via an example on the esr38 branch. Problem is that the parent doesn't need to be in the sequence but can also be a relbranch, so we will be driven away. Instead of using the hg rest api we should make use of Treeherder. We can get changesets before our wanted revision via that URL:

https://treeherder.mozilla.org/api/project/mozilla-esr38/resultset/?count=10&filter-tier=2&filter-tier=3&full=true&tochange=43cd30a17135

Then we walk the array that long until we actually found a revision with builds.
